[WebGPU] [Style] Inline members of Texture::m_descriptor

litherum opened the pull request:
#### 81eb82843a2a882d22e932d3a773a679f7cf5ee7
<pre>
[WebGPU] [Style] Inline members of Texture::m_descriptor
<a href="https://bugs.webkit.org/show_bug.cgi?id=251325">https://bugs.webkit.org/show_bug.cgi?id=251325</a>
rdar://104786027

Reviewed by Tadeu Zagallo.

This is a purely stylistic change, to match our implementation to the nouns-and-verbs
that the spec uses. The spec for GPUTexture says it has internal slots for each of
the fields individually, rather than retaining the whole descriptor. I think it&apos;s
a good idea to match the spec, just stylistically.

No tests because there is no behavior change.

* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::validateCopyBufferToTexture):
(WebGPU::CommandEncoder::copyBufferToTexture):
(WebGPU::validateCopyTextureToBuffer):
(WebGPU::CommandEncoder::copyTextureToBuffer):
(WebGPU::validateCopyTextureToTexture):
(WebGPU::CommandEncoder::copyTextureToTexture):
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::validateWriteTexture):
(WebGPU::Queue::writeTexture):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::width const):
(WebGPU::Texture::height const):
(WebGPU::Texture::depthOrArrayLayers const):
(WebGPU::Texture::mipLevelCount const):
(WebGPU::Texture::sampleCount const):
(WebGPU::Texture::dimension const):
(WebGPU::Texture::format const):
(WebGPU::Texture::usage const):
(WebGPU::Texture::descriptor const): Deleted.
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::Texture):
(WebGPU::Texture::resolveTextureViewDescriptorDefaults const):
(WebGPU::Texture::arrayLayerCount const):
(WebGPU::Texture::validateCreateView const):
(WebGPU::Texture::createView):
(WebGPU::Texture::logicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::physicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::validateImageCopyTexture):
(WebGPU::Texture::validateTextureCopyRange):

Canonical link: <a href="https://commits.webkit.org/259565@main">https://commits.webkit.org/259565@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/eda1e65eca5ce550452a4c87bef6bb8a74278028

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105179 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14262 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38062 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114438 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/15407 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5179 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97493 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/114386 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/110935 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/11903 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94895 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/39404 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/93770 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26530 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/81070 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/7593 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27889 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/7688 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/4466 "Found 1 new test failure: fast/images/avif-as-image.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/13740 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47444 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6593 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/9476 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WebGPU/Texture.h:
```diff
@@ -84,7 +84,15 @@ class Texture : public WGPUTextureImpl, public RefCounted<Texture> {
     id<MTLTexture> texture() const { return m_texture; }
-    const WGPUTextureDescriptor& descriptor() const { return m_descriptor; }
+
+    uint32_t width() const { return m_width; }
+    uint32_t height() const { return m_height; }
+    uint32_t depthOrArrayLayers() const { return m_depthOrArrayLayers; }
+    uint32_t mipLevelCount() const { return m_mipLevelCount; }
+    uint32_t sampleCount() const { return m_sampleCount; }
+    WGPUTextureDimension dimension() const { return m_dimension; }
+    WGPUTextureFormat format() const { return m_format; }
+    WGPUTextureUsageFlags usage() const { return m_usage; }
```
What does this really win us?

litherum replied:
Just matching the spec. https://gpuweb.github.io/gpuweb/#texture-interface

tadeuzagallo replied:
If you think that's worth it, I'll trust your judgement

litherum committed:
[WebGPU] [Style] Inline members of Texture::m_descriptor
https://bugs.webkit.org/show_bug.cgi?id=251325
rdar://104786027

Reviewed by Tadeu Zagallo.

This is a purely stylistic change, to match our implementation to the nouns-and-verbs
that the spec uses. The spec for GPUTexture says it has internal slots for each of
the fields individually, rather than retaining the whole descriptor. I think it's
a good idea to match the spec, just stylistically.

No tests because there is no behavior change.

* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::validateCopyBufferToTexture):
(WebGPU::CommandEncoder::copyBufferToTexture):
(WebGPU::validateCopyTextureToBuffer):
(WebGPU::CommandEncoder::copyTextureToBuffer):
(WebGPU::validateCopyTextureToTexture):
(WebGPU::CommandEncoder::copyTextureToTexture):
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::validateWriteTexture):
(WebGPU::Queue::writeTexture):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::width const):
(WebGPU::Texture::height const):
(WebGPU::Texture::depthOrArrayLayers const):
(WebGPU::Texture::mipLevelCount const):
(WebGPU::Texture::sampleCount const):
(WebGPU::Texture::dimension const):
(WebGPU::Texture::format const):
(WebGPU::Texture::usage const):
(WebGPU::Texture::descriptor const): Deleted.
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::Texture):
(WebGPU::Texture::resolveTextureViewDescriptorDefaults const):
(WebGPU::Texture::arrayLayerCount const):
(WebGPU::Texture::validateCreateView const):
(WebGPU::Texture::createView):
(WebGPU::Texture::logicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::physicalMiplevelSpecificTextureExtent):
(WebGPU::Texture::validateImageCopyTexture):
(WebGPU::Texture::validateTextureCopyRange):

Canonical link: https://commits.webkit.org/259565@main
